packages/common/util/src/assign.ts

```diff
@@ -8,7 +8,7 @@ import { invariant } from '@dxos/invariant';
  * Initialize a deeply nested object.
  * @returns The value of the prop after assignment.
  */
-export const assignDeep = <T>(doc: any, path: (string | number)[], value: T): T => {
+export const assignDeep = <T>(doc: any, path: readonly (string | number)[], value: T): T => {
```

Comment:
doesn't this mean that you can _only_ pass readonly paths? won't this make it generally hard to use? (i.e. require casting a lot of things when passing in)

Reply:
It goes the other way around: `string[]` is assignable to `readonly string[]` but not vice-versa